src/wallet/reconstruct_seed_flow.c

```diff
@@ -281,25 +279,68 @@ bool reconstruct_seed_flow(const uint8_t *wallet_id, uint8_t *seed_out) {
   }
 
   // Run flow till it reaches a completion state
-  reconstruct_state_e current_state = PASSPHRASE_INPUT;
+  reconstruct_state_e current_state = init_state;
   while (1) {
     reconstruct_state_e next_state =
-        reconstruct_seed_handler(current_state, seed_out);
+        reconstruct_seed_handler(current_state, secret_out);
 
     current_state = next_state;
     if (COMPLETED <= current_state) {
       break;
     }
   }
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+bool reconstruct_seed_flow(const uint8_t *wallet_id, uint8_t *seed_out) {
+  if ((NULL == wallet_id) || (NULL == seed_out)) {
+    return false;
+  }
+
+  uint8_t secret[BLOCK_SIZE] = {0};
+  uint8_t result = false;
+
+  if (COMPLETED ==
+      handle_secret_reconstruction(wallet_id, secret, PASSPHRASE_INPUT)) {
+    get_seed_from_secret(secret, seed_out);
+    result = true;
+  }
 
   // Clear confidential data irrespective of the result of the flow
   memzero(&wallet, sizeof(wallet));
   memzero(&wallet_shamir_data, sizeof(wallet_shamir_data));
   memzero(&wallet_credential_data, sizeof(wallet_credential_data));
 
-  if (COMPLETED == current_state) {
-    return true;
+  return result;
+}
+
+uint8_t reconstruct_mnemonics_flow(
+    const uint8_t *wallet_id,
+    char mnemonic_list[MAX_NUMBER_OF_MNEMONIC_WORDS]
+                      [MAX_MNEMONIC_WORD_LENGTH]) {
+  if ((NULL == wallet_id) || (NULL == mnemonic_list)) {
+    return 0;
   }
 
-  return false;
-}
+  uint8_t secret[BLOCK_SIZE] = {0};
+  uint8_t result = 0;
+
+  if (COMPLETED == handle_secret_reconstruction(wallet_id, secret, PIN_INPUT)) {
+    mnemonic_clear();
+    const char *mnemo =
+        mnemonic_from_data(secret, wallet.number_of_mnemonics * 4 / 3);
+    uint16_t single_line_mnemonics_length =
+        strnlen(mnemo, MAX_NUMBER_OF_MNEMONIC_WORDS * MAX_MNEMONIC_WORD_LENGTH);
+    __single_to_multi_line(mnemo, single_line_mnemonics_length, mnemonic_list);
+    result = wallet.number_of_mnemonics;
```

Comment:
We are potentially making a double copy of mnemonics.. can you re-check if this is required? I think this copy of mnemonics is not getting cleared as well.. 

We can move this logic in `view_seed_flow` and prevent double copy and therefore risk of missing out anything

Reply:
It is not a double copy as the of mnemonics as `mnemo` is a pointer for a static variable with single line mnemonics, and we need to convert the mnemonics to multi-line, we wouldn't want to change the pointer returned by `mnemonic_from_data`.

Reply:
To the second comment we can move this to view_seed_flow too, but it would still need a function that returns the secret for mnemonics construction, I think it's better to keep that logic abstracted as is.